### ISSUE TITLE
Hot reload doesn't work

### ISSUE THREAD

**Issue by nikhilyeole1** [Microsoft/TypeScript-React-Starter#50]
Hot reload works properly with just create-react-app but doesn't work with react-scripts-ts.

I have included following in index.tsx

```
if ((module as any).hot) {
  (module as any).hot.accept('./App', () => {
    const NextApp = require('./App').default;
    ReactDOM.render(<NextApp />, document.getElementById('root'));
  });
}
```

Since I am using create-react-app that uses webpack-2, I have tried following also:

```
if ((module as any).hot) {
  (module as any).hot.accept('./App', () => {
    ReactDOM.render(<App />, document.getElementById('root'));
  });
}
```
But still it doesn't work. CSS hot reload works btw. I also converted my App component to pure function instead of class but to no avail.

Following are the versions:
create-react-app - 1.3.3
react-scripts-ts - 2.4.0

Please help!

**Comment by NeekSandhu** [comment]
Did you try this one https://github.com/gaearon/react-hot-loader ?

**Comment by nikhilyeole1** [comment]
@NeekSandhu Yes I know that project. But for that I will need to eject :(

**Comment by NeekSandhu** [comment]
Well a fragile and dirty way would be go inside `node_modules/react-scripts` and make manual adjustments (definitely fragile), but then again.

`create-react-app` team doesn't look in the state to have this natively integrated anytime soon https://github.com/facebookincubator/create-react-app/pull/110

**Comment by Timer** [comment]
I would advise against ejecting to use `react-hot-loader`, it's simply too unstable.
You should stick to the defaults `create-react-app` provides until we ship hot reloading by default.

Hot reloading is currently blocked by some trivial project bugs and a bug in chrome, but coming soon: https://github.com/facebookincubator/create-react-app/pull/2304.

We're hopefully also [adding support for TypeScript soon](https://github.com/facebookincubator/create-react-app/pull/2815), so users no longer need to use the community-maintained `react-scripts-ts` fork.

**Comment by nikhilyeole1** [comment]
Thanks @Timer for clarification. Looking forward to get the TS support soon 👍

**Comment by nikhilyeole1** [comment]
@Timer Found the solution. I replaced ts-loader with awesome-typescript-loader and hot reloading works now. I had to eject to debug/try things. I will use https://github.com/timarney/react-app-rewired to customize my project to use awesome-typescript-loader without ejecting. 👍

**Comment by Timer** [comment]
Sounds good, @nikhilkh! While we don't promise `react-app-rewired` won't ever break, we try our best to remain compatible with it -- so maybe expect a few breaks every once and awhile.

**Comment by VRspace4** [comment]
Any updates on integrating hot module reloading into the create-react-app? Thanks.

**Comment by jackson-** [comment]
Agreed @VRspace4. Is this something we should see implemented soon?
